crates/ide-assists/src/handlers/bool_to_enum.rs

```diff
@@ -93,27 +94,36 @@ struct BoolNodeData {
 fn find_bool_node(ctx: &AssistContext<'_>) -> Option<BoolNodeData> {
     let name: ast::Name = ctx.find_node_at_offset()?;
 
-    if let Some(let_stmt) = name.syntax().ancestors().find_map(ast::LetStmt::cast) {
-        let bind_pat = match let_stmt.pat()? {
-            ast::Pat::IdentPat(pat) => pat,
-            _ => {
-                cov_mark::hit!(not_applicable_in_non_ident_pat);
-                return None;
-            }
-        };
-        let def = ctx.sema.to_def(&bind_pat)?;
+    if let Some(ident_pat) = name.syntax().ancestors().find_map(ast::IdentPat::cast) {
+        let def = ctx.sema.to_def(&ident_pat)?;
         if !def.ty(ctx.db()).is_bool() {
             cov_mark::hit!(not_applicable_non_bool_local);
             return None;
         }
 
-        Some(BoolNodeData {
-            target_node: let_stmt.syntax().clone(),
-            name,
-            ty_annotation: let_stmt.ty(),
-            initializer: let_stmt.initializer(),
-            definition: Definition::Local(def),
-        })
+        let local_definition = Definition::Local(def);
+        match ident_pat.syntax().ancestors().find_map(Either::<ast::Param, ast::LetStmt>::cast)? {
+            Either::Left(param) => Some(BoolNodeData {
+                target_node: param.syntax().clone(),
+                name,
+                ty_annotation: param.ty(),
+                initializer: None,
+                definition: local_definition,
+            }),
+            Either::Right(let_stmt) => match let_stmt.pat()? {
+                ast::Pat::IdentPat(_) => Some(BoolNodeData {
+                    target_node: let_stmt.syntax().clone(),
+                    name,
+                    ty_annotation: let_stmt.ty(),
+                    initializer: let_stmt.initializer(),
+                    definition: local_definition,
+                }),
+                _ => {
+                    cov_mark::hit!(not_applicable_in_non_ident_pat);
```

Comment:
Let's remove the `cov_mark` for this, this is covered by the proposed changes above (via the `?` bubbling), otherwise this just adds noise to the code imo.